.github/workflows/publish-on-space.yml

```
@@ -0,0 +1,21 @@
+name: Publish on Space
+
+on:
+  push:
+    branches: [master]
+
+jobs:
+  publish:
+    name: publish on space
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: SetUp Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: '11'
+      - name: Add properties and publish
+        run: |
+          echo publishingVersion=${{ github.sha }} | head -c 26 >> gradle.properties
```

Here we depend on the length of "publishingVersion=" and it's error-prone. Let's use just the length of the hash we want to use: `8`. So I guess it will be something like:

```suggestion
          publishingVersion=${{ github.sha }} | head -c 8
          echo publishingVersion=$publishingVersion >> gradle.properties
```

---

I can just add zero terminated row with version
```shell
echo publishingVersion= | head -c 18 -z >> gradle.properties
echo ${{ github.sha }}  | head -c 8 >> gradle.properties
```

---

fixed with `echo -n`